src/main/kotlin/com/autonomousapps/internal/utils/regex.kt

```diff
@@ -24,8 +24,14 @@ internal fun List<AnnotationNode>?.annotationTypes(): Set<String> {
 
   forEach { anno ->
     types.add(anno.desc)
-    anno.values.orEmpty().filterIsInstance<Type>().forEach { type ->
-      types.add(type.descriptor)
+
+    anno.values.orEmpty().filter {
+      it is Type || it is ArrayList<*>
+    }.forEach { value ->
+      when (value) {
+        is Type -> types.add(value.descriptor)
+        is ArrayList<*> -> value.filterIsInstance<Type>().forEach { type -> types.add(type.descriptor) }
```

Comment:
Just checking, could this be a `List<List<...>>`?

Also, must it be an `ArrayList`? Why not just `List`?

Reply:
The type comes through as `ArrayList`. A naive check for `List` fails but I can cast and check if it succeeds as an option?

Reply:
I'm just wondering if there's some guarantee that it's always an `ArrayList`. For obvious reasons I prefer to use a more general type whenever possible.

Reply:
Ah I see. Yeah you're right, List work's here and matches what ASM expects: https://asm.ow2.io/javadoc/org/objectweb/asm/tree/AnnotationNode.html#values